apps/web/src/components/AgentDetail/AgentDetailLayout.tsx

```diff
@@ -0,0 +1,326 @@
+"use client";
+
+import {
+  useEffect,
+  useRef,
+} from "react";
+
+import {
+  BarChart3,
+  ChevronDown,
+  Clock,
+  Download,
+  FileDown,
+  FileText,
+  Pencil,
+  Play,
+  Upload,
+  User,
+} from "lucide-react";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+
+import { Button } from "@/components/Button";
+import { ExperimentalBadge } from "@/components/ExperimentalBadge";
+import { AgentBadges } from "@/components/AgentBadges";
+
+import type { AgentDetailProps } from "./types";
+import {
+  exportAgentAsJson,
+  exportAgentAsMarkdown,
+  exportAgentAsYaml,
+} from "./utils";
+import { useState } from "react";
+
+export function AgentDetailLayout({
+  agent,
+  isOwner = false,
+  isAdmin = false,
+  children,
+}: AgentDetailProps & { children: React.ReactNode }) {
+  const pathname = usePathname();
+  const [exportDropdownOpen, setExportDropdownOpen] = useState(false);
+  const [copySuccess, setCopySuccess] = useState(false);
+  const [exportType, setExportType] = useState<"JSON" | "Markdown" | "YAML">("JSON");
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  // Close dropdown when clicking outside
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setExportDropdownOpen(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [setExportDropdownOpen]);
+
+  const handleExportJson = () =>
+    exportAgentAsJson(agent, setExportType, setCopySuccess);
+  const handleExportMarkdown = () =>
+    exportAgentAsMarkdown(agent, setExportType, setCopySuccess);
+  const handleExportYaml = () =>
+    exportAgentAsYaml(agent, setExportType, setCopySuccess);
+
+  // Determine active tab from pathname using more robust segment matching
+  const getActiveTab = () => {
+    // Split pathname and get the last segment
+    const segments = pathname.split('/').filter(Boolean);
+    const lastSegment = segments[segments.length - 1];
+    
+    // Check if the last segment is a known tab
+    const validTabs = ['overview', 'details', 'evals', 'jobs', 'test', 'batches', 'export', 'versions'];
+    if (validTabs.includes(lastSegment)) {
+      return lastSegment;
+    }
+    
+    // Default to overview if we're on the agent base path
+    return 'overview';
+  };
+
+  const activeTab = getActiveTab();
+
+  return (
```

Comment:
this statement is huge. Clean it up by separating out parts of it into smaller components and similar